packages/dds/merge-tree/src/test/revertibles.spec.ts

```diff
@@ -0,0 +1,240 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import {
+    appendToMergeTreeDeltaRevertibles,
+    MergeTreeDeltaRevertible,
+    revertMergeTreeDeltaRevertibles,
+} from "../revertibles";
+import { createRevertDriver } from "./testClient";
+import { createClientsAtInitialState, TestClientLogger } from "./testClientLogger";
+
+describe("MergeTree.Revertibles", () => {
+    it("revert insert", () => {
+        const clients = createClientsAtInitialState(
+            { initialState: "123", options: { mergeTreeUseNewLengthCalculations: true } },
+            "A", "B");
+        const logger = new TestClientLogger(clients.all);
+        let seq = 0;
+        const ops: ISequencedDocumentMessage[] = [];
+
+        const clientB_Revertibles: MergeTreeDeltaRevertible[] = [];
+        // the test logger uses these callbacks, so preserve it
+        const old = clients.B.mergeTreeDeltaCallback;
+        const clientBDriver = createRevertDriver(clients.B);
+        clientBDriver.submitOpCallback = (op) => ops.push(clients.B.makeOpMessage(op, ++seq));
```

Comment:
any reason to not build in some of the callback boilerplate into the test revert driver? seems like it's reasonably within its realm of concerns and tests here generally all do the same thing

Reply:
i've thought about it, but i'm ok with some redundancy in test if it makes the cases more self contained. it really easy to start building layers in tests that make it harder to use and debug later.

Reply:
fair enough. Delaying abstraction when not positive you've got it right is a good rule of thumb...